### ISSUE TITLE
fish_vi_cursor -h spews errors, not friendly

### ISSUE THREAD

**Issue by floam** [fish-shell/fish-shell#3266]
- [x] Have you checked if problem occurs with [fish 2.3.1](/fish-shell/fish-shell/releases/tag/2.3.1)?
- [x] Tried fish without third-party customizations _(check `sh -c 'env HOME=$(mktemp -d) fish'`)_?

**fish version installed** _(`fish --version`)_: 2.3.1

**OS/terminal used**: OS X
Someone might be curious how fish_vi_cursor is supposed to be used. It's not prefixed (and not hooked up by default currently).
## Reproduction steps

Hard to explain? Link to [a screen recording](https://asciinema.org)
1. Launch fish
2. `> fish_vi_cursor -h`
## Expected results

Show help or exit 0 after setting up the cursor.
## Actual results

No help or results, but it does exit 0. However it seems all binds spew errors after.
1. `> fish_vi_cursor -h`
2. `> [ENTER]`

```
>Variables may not be used as commands. In fish, please define a function or use 'eval $$varname'.
- (line 6):                $$varname
                           ^
in function 'fish_vi_cursor_handle'
    called on standard input
    with parameter list 'VARIABLE SET fish_bind_mode'
```

[![asciicast](https://asciinema.org/a/3bct27l7oehxy58g0orfcgr4n.png)](https://asciinema.org/a/3bct27l7oehxy58g0orfcgr4n)

**Comment by floam** [comment]
When used interactively, it should print to the terminal that the key bindings have changed, the cursor changed (if it changed), and that indicator was added and what it means. It should tell the user how to go back.

**Comment by krader1961** [comment]
This is a dup of issue #3472 which was fixed by commit 708f80d8.
